File: 브론즈2/Java/B0806/src/B4583.java
import java.util.Scanner;

public class B4583 {
    public static void main(String[] args) {
        char[] reflect = {'b', 'd', 'p', 'q', 'i', 'o', 'v', 'w', 'x'};

        while (true) {
            Scanner scanner = new Scanner(System.in);
            String word = scanner.nextLine();

            if (word.equals("#")) {
                break;
            }

            StringBuilder result = new StringBuilder();

            for (char alphabet : word.toCharArray()) {
                if (new String(reflect).indexOf(alphabet) == -1) {
                    result = new StringBuilder("INVALID");
                    break;
                }

                else {
                    if (alphabet == 'b') {
                        result.append('d');
                    } else if (alphabet == 'd') {
                        result.append('b');
                    } else if (alphabet == 'p') {
                        result.append('q');
                    } else if (alphabet == 'q') {
                        result.append('p');
                    }
                    else {
                        result.append(alphabet);
                    }
                }
            }

            if (!result.toString().equals("INVALID")) {
                System.out.println(result.reverse().toString());
            }
            else {
                System.out.println(result);
            }
        }
    }
}

File: 브론즈2/PY/0918/B4583.py
reflect = ['b', 'd', 'p', 'q', 'i', 'o', 'v', 'w', 'x']

while True:
    word = input()

    if word == '#':
        break
    result = ''

    for alphabet in word:
        if alphabet not in reflect:
            result = "INVALID"
            break
        else:
            if alphabet == 'b':
                result += 'd'
            elif alphabet == 'd':
                result += 'b'
            elif alphabet == 'p':
                result += 'q'
            elif alphabet == 'q':
                result += 'p'
            else:
                result += alphabet
    if result != 'INVALID':
        print(result[::-1])
    else:
        print(result)
